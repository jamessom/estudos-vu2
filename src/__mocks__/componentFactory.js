import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from '@/router/index';

export default function componentFactory(options = {}) {
  return {
    wrapper: () => {
      const { component, componentOptions, withRouter } = options;
      const localvue = createLocalVue();

      if (withRouter) {
        localvue.use(VueRouter);

        return shallowMount(component, {
          localvue,
          ...componentOptions,
        });
      }

      return shallowMount(component, {
        localvue,
        ...componentOptions,
      });
    },
  };
}
