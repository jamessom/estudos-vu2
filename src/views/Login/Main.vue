<template>
  <div class="columns is-centered is-vcentered">
    <div class="column is-3 is-offset-0">
      <div class="container">
        <div class="has-text-centered">
            <img
              class="login-logo"
              :src="require('@/assets/images/logo_horizontal.png')"
            />
        </div>

        <input-base label="E-mail" inputType='email' v-model="email" />
        <input-base label="Senha" inputType='password' v-model="password" />

        <div class="has-text-centered is-pulled-right">
          <a class="button is-vcentered is-primary is-outlined">Login</a>
        </div>

        <div class="is-vcentered is-pulled-left">
          <a href="/register">
            Você não tem uma conta? <br>
            Crie uma agora
          </a>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import InputBase from '@/components/InputBase.vue';

export default {
  name: 'Login',
  data() {
    return {
      email: '',
      password: '',
    };
  },
  components: {
    InputBase,
  },
};
</script>
