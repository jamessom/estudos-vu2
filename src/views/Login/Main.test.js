import componentFactory from '@/__mocks__/componentFactory';
import Login from './Main.vue';

describe('src/views/Login/Main.vue', () => {
  const factory = componentFactory({
    component: Login,
  });

  it('should renders correctly', () => {
    expect(factory.wrapper().element).toMatchSnapshot();
  });
});
