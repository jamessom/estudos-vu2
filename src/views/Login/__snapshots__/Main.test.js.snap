// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/views/Login/Main.vue should renders correctly 1`] = `
<div
  class="columns is-centered is-vcentered"
>
  <div
    class="column is-3 is-offset-0"
  >
    <div
      class="container"
    >
      <div
        class="has-text-centered"
      >
        <img
          class="login-logo"
          src=""
        />
      </div>
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="email"
        label="E-mail"
        value=""
      />
       
      <input-base-stub
        controlclass="control"
        inputclass="input"
        inputtype="password"
        label="Senha"
        value=""
      />
       
      <div
        class="has-text-centered is-pulled-right"
      >
        <a
          class="button is-vcentered is-primary is-outlined"
        >
          Login
        </a>
      </div>
       
      <div
        class="is-vcentered is-pulled-left"
      >
        <a
          href="/register"
        >
          
          Você não tem uma conta? 
          <br />
          
          Crie uma agora
        
        </a>
      </div>
    </div>
  </div>
</div>
`;
